exemplars_parent/exemplars/src/main/java/io/prometheus/client/exemplars/api/CounterExemplarSampler.java

```diff
@@ -0,0 +1,18 @@
+package io.prometheus.client.exemplars.api;
+
+/**
+ * Exemplar sampler for counter metrics.
+ */
+public interface CounterExemplarSampler {
+
+  /**
+   * @param increment     the value added to the counter on this event
+   * @param newTotalValue the new total counter value after adding the increment.
```

Comment:
Why would you care about this? Increments should be independent.

Reply:
I'm not sure about this. Here's an example that might be a valid use case: A counter representing a buffer size. If the buffer can only grow, a counter should be a valid metric. A user might decide to sample exemplars only when the buffer has exceeded a certain size, thus the exemplar sampler needs the total value.

I'm not saying that this will be common, in most cases users won't need the total value. However, if we remove it from the API there will be no chance to use it even if there are rare cases where it makes sense.

Anyway, I don't have a strong opinion on this. If you still think removing it would be better I'm happy to remove it.

Reply:
The interesting thing about a counter is how quickly it goes up, the absolute value doesn't matter.

>  A counter representing a buffer size

That's a gauge, as you care about the absolute value.